spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/AbstractSpannerIntegrationTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.test;
+
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseId;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerPersistentEntity;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerPersistentEntityImpl;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.data.util.TypeInformation;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * <p>
+ * For Spanner integration tests, you will need to have an instance predefined, everything
+ * else is generated. The instance by default is "integration-instance", which can be
+ * overriden in <code>src/test/resources/application-test.properties</code>, the property:
+ * <code>test.integration.spanner.instance</code>
+ * </p>
+ * <p>
+ * Within the <code>integration-instance</code> instance, there is a single database for
+ * tests, <code>integration-db</code>. This is automatically created, if doesn't exist.
+ * The tables are generated to have a unique postfix, which is updated on the entity
+ * annotations as well dynamically to avoid collisions of multiple parallel tests running
+ * against the same instance.
+ * </p>
+ *
+ * @author Balint Pato
+ */
+@ContextConfiguration(classes = { IntegrationTestConfiguration.class })
```

Comment:
Why aren't we using `@SpringBootTest` instead?

Reply:
I didn't want to depend on Spring Boot for integration testing.

Reply:
Why not? Could it help us eliminate the `IntegrationTestConfiguration` class?
If you don't want to depend on SB for integration tests but need to inject things into a context, have you considered using an `ApplicationContextRunner`?

Reply:
That does not make too much sense to restrict yourself without Spring Boot Test, when the main code is already Spring Boot based.

Right, you might not need the whole Spring Boot auto-configuration for testing. But you can consider really to use the mentioned `ApplicationContextRunner` or consider to introduce an appropriate testing `slice`: https://docs.spring.io/spring-boot/docs/2.0.0.RELEASE/reference/htmlsingle/#boot-features-testing-spring-boot-applications-testing-autoconfigured-jpa-test